compiler/plugins/input/TOSA/InputConversion/test/apply_pdll_patterns_tosa.mlir

```diff
@@ -0,0 +1,54 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(iree-preprocessing-apply-pdll-patterns{patterns-file=%p/tosa.pdll})" %s | FileCheck %s
+
+// CHECK-LABEL:   stream.executable private @mlp_external_f32_f32_f32_i32_i32_i32_i1_executable
```

Comment:
it is supposed to be some sort of mangled name. So maybe worth leaving it in as is.

Reply:
I agree with Scott's meta-point here in that this is quite an involved CHECK file. I see this same kind of thing tested across multiple input types as well. Could we deduplicate a bit?

Reply:
I like to have CHECKs for everything that is generated. So that it actually serves as a check to match what is being generated. Its just CHECK strings, some duplication on a few files seem OK?

Reply:
Torch and TOSA seem to be generating essentially the exact same verbose checks for the same rewrite, meaning we need to update it in two places if there are IR changes. It's not a huge deal but I thought the rule of thumb was to try to keep tests distinct.

Reply:
https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices

> Tests should be minimal, and only check what is absolutely necessary.
This means that anything in the output that is not core to the functionality that you are testing should not be present in a CHECK line.

Reply:
If you check everything that is generated, you've written a change detector test that is a chore to keep updated.

Reply:
Would It be better to check for the call to `@mlp_external...` + the final output?

Reply:
Ok... ill defer to you guys to provide guidance here, but I dont consider lit tests to be anything more than change detector :D

Reply:
Right, they are change detectors by construction, but the idea is to make the detected changes meaningful. In this case I think it would be fine to drop either the Torch or TOSA lit test because e2e is already tested in the plugin samples for each input type.

Reply:
The Torch and TOSA lit test was just to show, you can do either, but yeah OK duplication doesnt make sense. @IanWood1  maybe just check that this is producing the dispatch. The PDL scripts are calling the same function and that is checking the generated code.

Reply:
Okay, I removed the Torch and TOSA tests reduced the amounts of checks in plugins

Reply:
Thanks! Although I didn't mean removing the checks for all of them because it looked like it was testing the actual native rewrite pattern as well, mainly just that we didn't need the same set of verbose checks in three places. Adding back the linalg one maybe seems reasonable to me if you want.